drivers/hmis/app/models/hmis/ce/referral.rb

```diff
@@ -139,6 +145,41 @@ def self.apply_filters(input)
       Hmis::Filter::CeReferralFilter.new(input).filter_scope(self)
     end
 
+    # Returns IDs of Referrals where there are completed steps assigned to
+    # a swimlane that the provided user participates in on that specific referral.
+    #
+    # These referrals are included in visibility for users with "can view own referrals" permission.
+    # This allows users to see referrals that they have been added to retroactively, even if the steps
+    # assigned to "their" swimlane have all been completed.
+    #
+    # @return [Array<Integer>] Array of referral IDs
+    def self.referral_ids_for_user_with_completed_swimlane_steps(user)
+      # Get all referral participants for this user, with their workflow instance IDs
+      participants = user.ce_referral_participants.joins(:referral).
+        select(:referral_id, :swimlane_id, 'ce_referrals.workflow_instance_id').
+        distinct
+
+      # Get all workflow instance IDs that this user is participating in
+      instance_ids = participants.map(&:workflow_instance_id)
+      return unless instance_ids.any? # user is not participating in any referrals
```

Comment:
return consistent array?
```suggestion
      return [] unless instance_ids.any? # user is not participating in any referrals
```